File: src/main/java/me/melijn/dhs/components/SwitchComponent.java
package me.melijn.dhs.components;

public class SwitchComponent extends Component {

    private int id;
    private boolean on;
    private int onCode;
    private int offCode;

    public SwitchComponent(String name, Location location, int id, boolean state, int onCode, int offCode) {
        super(name, location, ComponentType.SWITCH);
        this.id = id;
        this.on = state;
        this.onCode = onCode;
        this.offCode = offCode;
    }

    public int getOffCode() {
        return offCode;
    }

    public int getOnCode() {
        return onCode;
    }

    public boolean isOn() {
        return on;
    }

    public void setOn(boolean on) {
        this.on = on;
    }

    public int getId() {
        return id;
    }
}

File: src/main/kotlin/me/melijn/dhs/objects/components/SwitchComponent.kt
package me.melijn.dhs.objects.components

import com.fasterxml.jackson.databind.node.ObjectNode
import me.melijn.dhs.rest.OBJECT_MAPPER

class SwitchComponent(name: String, location: Location, val id: Int, var isOn: Boolean, val onCode: Int, val offCode: Int) : Component(name, location, ComponentType.SWITCH) {

    override fun toObjectNode(): ObjectNode? {
        return OBJECT_MAPPER.createObjectNode()
            .put("id", id)
            .put("state", isOn)
    }
}